src/libraries/System.Private.CoreLib/src/System/IO/DerivedFileStreamStrategy.cs

```diff
@@ -36,7 +49,14 @@ public override long Position
 
         internal override string Name => _strategy.Name;
 
-        internal override SafeFileHandle SafeFileHandle => _strategy.SafeFileHandle;
+        internal override SafeFileHandle SafeFileHandle
+        {
+            get
+            {
+                _fileStream.Flush(false);
```

Comment:
Why do we need to do this for derived types? 
Is this something that we should be doing on other/all strategies?

Reply:
> Why do we need to do this for derived types?

This is a `FileStream` contract that is guarded by unit tests. We have to call `Flush(false)` for derived types.
I think that it's because somebody might want to create another `FileStream` out of the handle, so before they do it we need to flush any buffered data to disk.

```cs
FileStream fs1 = File.Open("path");
fs1.WriteByte(1); // buffered, not written to the disk

FileStream fs2 = new FileStream(fs1.SafeFileHandle);
Assert.True(1 == fs1.Position == fs2.Position);
```

> Is this something that we should be doing on other/all strategies?

Yes, and the buffering strategy does that:

https://github.com/dotnet/runtime/blob/e8d6f776ba0bf502f752826ca94c6f0fbfcf8b1d/src/libraries/System.Private.CoreLib/src/System/IO/BufferedFileStreamStrategy.cs#L44-L51

We have a lot of unit tests that verify that (I know because I've learned about this contract after getting them red)